source/controllers/ListingController.ts

```diff
@@ -215,14 +215,20 @@ listing.interface('/create', async (req, res) => {
 	if (res.locals.user instanceof User) {
 		const sections = await Section.findAll({ include: [Subsection] });
 		let sectioncount = 0;
+		const secId = req.query.sectionId;
+		const subsecId = req.query.subsectionId;
 		sections.forEach(element => {
 			if (element.id >= sectioncount && element.id !== null && element.id !== undefined) {
 				sectioncount = element.id;
 			}
 		});
 		if (res.locals.user !== undefined && res.locals.user !== null) {
 			const user: User = res.locals.user;
-			res.render('pages/listing/create', { constants: headerConstants, userstatus_name: res.locals.user.username, userstatus_page: `/user/profile/${user.id}`, sections, sectioncount });
+			if (secId !== null && secId !== undefined && subsecId !== null && subsecId !== undefined) {
+				res.render('pages/listing/create', { constants: headerConstants, userstatus_name: res.locals.user.username, userstatus_page: `/user/profile/${user.id}`, sections, sectioncount, currentsection: secId, currentsubsection: subsecId, createfromexistsubcat: 'true' });
+			} else {
+				res.render('pages/listing/create', { constants: headerConstants, userstatus_name: res.locals.user.username, userstatus_page: `/user/profile/${user.id}`, sections, sectioncount, currentsection: 'null', currentsubsection: 'null', createfromexistsubcat: 'false' });
+			}
```

Comment:
You define a variable user, yet keep using res.locals.user everywhere. And move the user variable before the if statement and check `(user!==undefined && user!==null)` instead of `(res.locals.user!==`....